src/main/java/novamachina/exnihilosequentia/common/datagen/ExNihiloRecipeGenerator.java

```diff
@@ -99,16 +99,66 @@ private void registerBarrels(Consumer<IFinishedRecipe> consumer) {
+                .key('-', Items.BIRCH_SLAB)
+                .setGroup(ExNihiloConstants.ModIds.EX_NIHILO_SEQUENTIA)
+                .addCriterion("has_walls", hasItem(Items.BIRCH_PLANKS))
+                .addCriterion("has_base", hasItem(Items.BIRCH_SLAB))
+                .build(consumer, createSaveLocation(ExNihiloBlocks.BARREL_BIRCH.getId()));
+        ShapedRecipeBuilder.shapedRecipe(ExNihiloBlocks.BARREL_DARK_OAK.get())
+                .patternLine("x x")
+                .patternLine("x x")
+                .patternLine("x-x")
+                .key('x', Items.DARK_OAK_PLANKS)
+                .key('-', Items.DARK_OAK_SLAB)
+                .setGroup(ExNihiloConstants.ModIds.EX_NIHILO_SEQUENTIA)
+                .addCriterion("has_walls", hasItem(Items.DARK_OAK_PLANKS))
+                .addCriterion("has_base", hasItem(Items.DARK_OAK_SLAB))
+                .build(consumer, createSaveLocation(ExNihiloBlocks.BARREL_DARK_OAK.getId()));
+        ShapedRecipeBuilder.shapedRecipe(ExNihiloBlocks.BARREL_JUNGLE.get())
+                .patternLine("x x")
+                .patternLine("x x")
+                .patternLine("x-x")
+                .key('x', Items.JUNGLE_PLANKS)
+                .key('-', Items.JUNGLE_SLAB)
+                .setGroup(ExNihiloConstants.ModIds.EX_NIHILO_SEQUENTIA)
+                .addCriterion("has_walls", hasItem(Items.JUNGLE_PLANKS))
+                .addCriterion("has_base", hasItem(Items.JUNGLE_SLAB))
+                .build(consumer, createSaveLocation(ExNihiloBlocks.BARREL_JUNGLE.getId()));
+        ShapedRecipeBuilder.shapedRecipe(ExNihiloBlocks.BARREL_SPRUCE.get())
+                .patternLine("x x")
+                .patternLine("x x")
+                .patternLine("x-x")
+                .key('x', Items.SPRUCE_PLANKS)
+                .key('-', Items.SPRUCE_SLAB)
+                .setGroup(ExNihiloConstants.ModIds.EX_NIHILO_SEQUENTIA)
+                .addCriterion("has_walls", hasItem(Items.SPRUCE_PLANKS))
+                .addCriterion("has_base", hasItem(Items.SPRUCE_SLAB))
+                .build(consumer, createSaveLocation(ExNihiloBlocks.BARREL_SPRUCE.getId()));
+        ShapedRecipeBuilder.shapedRecipe(ExNihiloBlocks.BARREL_OAK.get())
+                .patternLine("x x")
+                .patternLine("x x")
+                .patternLine("x-x")
+                .key('x', Items.OAK_PLANKS)
+                .key('-', Items.OAK_SLAB)
                 .setGroup(ExNihiloConstants.ModIds.EX_NIHILO_SEQUENTIA)
-                .addCriterion("has_walls", hasItem(ItemTags.PLANKS))
-                .addCriterion("has_base", hasItem(ItemTags.WOODEN_SLABS))
-                .build(consumer, createSaveLocation(ExNihiloBlocks.BARREL_WOOD.getId()));
+                .addCriterion("has_walls", hasItem(Items.OAK_PLANKS))
+                .addCriterion("has_base", hasItem(Items.OAK_SLAB))
+                .build(consumer, createSaveLocation(ExNihiloBlocks.BARREL_OAK.getId()));
```

Comment:
This could probably be simplified with a method, but we'll fix that later.

Reply:
you got it, i moved crucibles and sieves into their own method to see which ones are new, i also added the TODO to find them more easy